tableauserverclient/server/endpoint/sites_endpoint.py

```diff
@@ -59,7 +59,12 @@ def delete(self, site_id):
             raise ValueError(error)
         url = "{0}/{1}".format(self.baseurl, site_id)
         self.delete_request(url)
-        logger.info('Deleted single site (ID: {0})'.format(site_id))
+        # Deleting the site also signs you out on the Server.
+        # Here we check that you're deleting the site you are signed in to
+        # (Which is true now but may not always be) and then clear auth tokens
+        if site_id == self.parent_srv._site_id:
```

Comment:
The getter checks if you have a site_id, which we don't because we remove it in the DELETE call when we make it, and then complains that you aren't signed in.

```
    @property
    def site_id(self):
        if self._site_id is None:
            error = 'Missing site ID. You must sign in first.'
            raise NotSignedInError(error)
        return self._site_id
```